tests/oc_template_test.go

```diff
@@ -0,0 +1,331 @@
+							}
+						}
+					}
+				},
+				\"apiVersion\": \"kubevirt.io/v1alpha1\",
+				\"metadata\": {
+					\"labels\": {
+						\"kubevirt-ovm\": \"ovm-${NAME}\"
+					},
+					\"name\": \"${NAME}\"
+				}
+			}
+		],
+		\"apiVersion\": \"v1\",
+		\"parameters\": [
+			{
+				\"name\": \"NAME\",
+				\"description\": \"Name for the new VM\"
+			},
+			{
+				\"name\": \"MEMORY\",
+				\"value\": \"4096Mi\",
+				\"description\": \"Amount of memory\"
+			},
+			{
+				\"name\": \"CPU_CORES\",
+				\"value\": \"4\",
+				\"description\": \"Amount of cores\"
+			}
+		],
+		\"metadata\": {
+			\"labels\": {
+				\"miq.github.io/kubevirt-is-vm-template\": \"true\",
+				\"kubevirt.io/os\": \"fedora27\"
+			},
+			\"annotations\": {
+				\"description\": \"OCP KubeVirt Fedora 27 VM template\",
+				\"tags\": \"kubevirt,ocp,template,linux\"
+			},
+			\"console.openshift.io/iconClass\": \"icon-fedora\",
+			\"name\": \"fedora-vm-template\"
+		}
+	}"
+
+	Context("with oc command", func() {
+		var vmTemplateJsonFile = ""
+		var vmJsonFile = ""
+
+		BeforeEach(func() {
+			tests.SkipIfNoOc()
+			// login into oc CLI
+			out, err = RunOcCommand("login", "-u", ocUser)
+			Expect(err).ToNot(HaveOccurred())
+			// write testing vm template into file
+			vmTemplateJsonFile, err = WriteJson("testvm-template", vmTemplate)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			if vmTemplateJsonFile != "" {
+				// remove testing vm template
+				err = os.Remove(vmTemplateJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmTemplateJsonFile = ""
+			}
+			if vmJsonFile != "" {
+				// remove testing vm JSON file
+				err = os.Remove(vmJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmJsonFile = ""
+			}
+		})
+
+		It("should convert an offline vm template into JSON file", func() {
+			By("Converting offline vm template into JSON file via oc-process command")
+			out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
+			Expect(err).ToNot(HaveOccurred())
+			vmJsonFile, err = WriteJson(vmName, out)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should create an offline vm from a template", func() {
+			var message = ""
+			// probably there is a better way how to do this without duplicating the code
+			By("Converting offline vm template into JSON file via oc-process command")
+			out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
+			Expect(err).ToNot(HaveOccurred())
+			vmJsonFile, err = WriteJson(vmName, out)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating offline vm from JSON file via oc-create command")
+			out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
+			Expect(err).ToNot(HaveOccurred())
+			message = fmt.Sprintf("offlinevirtualmachine \"%s\" created", vmName)
+			Expect(out).To(Equal(message))
+
+			// this should be probably handled differently
+			time.Sleep(time.Second * 120)
```

Comment:
the `Eventually()` command can help here. Assuming the goal here is to wait for the VM to be running, something along these lines:

```
Eventually(func() bool {
   out, err = tests.RunOcCommand("look up that vm...")
   Expect(err).ToNot(HaveOccurred())
   return out == "???"
}, time.Duration(60)*time.Second).Should(Equal(true), "Timed out waiting for preset to appear")
```

Reply:
Note the func you define here does not need to return a bool. if an error object, integer, anything is more appropriate, that's fair game. Just make the Should() comparison match what you're looking for....

Reply:
Yes Eventually is the way to go ...

Is oc-create asynchronous? If not, you should immediatley have the chance to fetch the vm.